### PR TITLE
fixes invalid zip content

### PR DESCRIPTION
**Pull request by briangann**
* update to go 1.17
* update packages
* handle plugin id and version detection
* handle malformed zip files
* add testdata

### CODE REVIEW

File: pkg/cmd/plugincheck2/main.go [hunk 1]
```diff
@@ -15,6 +15,12 @@ import (
 	"gopkg.in/yaml.v2"
 )
 
+type FormattedOutput struct {
+	ID         string                           `json:"id"`
+	Version    string                           `json:"version"`
+	Diagnotics map[string][]analysis.Diagnostic `json:"plugin-validator"`
```
Comment (marcusolsson) [inline pkg/cmd/plugincheck2/main.go:21]:
```suggestion
	Diagnostics map[string][]analysis.Diagnostic `json:"plugin-validator"`
```

File: pkg/cmd/plugincheck2/quickmeta.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+// GetIDAndVersion Gets plugin id and version from extracted zip
+func GetIDAndVersion(archiveDir string) (*string, *string, error) {
+	fis, _ := ioutil.ReadDir(archiveDir)
```
Comment (marcusolsson) [inline pkg/cmd/plugincheck2/quickmeta.go:13]: Are we certain the archive directory exists here? If it doesn't, it'll panic on the next line.

File: pkg/cmd/plugincheck2/quickmeta.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+// GetIDAndVersion Gets plugin id and version from extracted zip
+func GetIDAndVersion(archiveDir string) (*string, *string, error) {
```
Comment (marcusolsson) [inline pkg/cmd/plugincheck2/quickmeta.go:12]: Why are we returning references to strings instead of values?

Comment (briangann) [reply pkg/cmd/plugincheck2/quickmeta.go:12]: doesn't have to, was just returning nil/nil/error vs "","",err

File: pkg/cmd/plugincheck2/quickmeta.go (hunk 2 above)
Comment (briangann) [reply pkg/cmd/plugincheck2/quickmeta.go:13]: good catch will update

Comment (briangann) [reply pkg/cmd/plugincheck2/quickmeta.go:25]: redid this section to handle more edge cases

File: pkg/cmd/plugincheck2/quickmeta.go (hunk 3 above)
Comment (briangann) [reply pkg/cmd/plugincheck2/quickmeta.go]: updated this also

**Review by marcusolsson — APPROVED** [review]

**Pull request merged**